src/components/Form.tsx

```diff
@@ -1,15 +1,48 @@
+import { useState } from 'react';
 import { useForm } from 'react-hook-form';
-import Typography from '@mui/material/Typography';
+import { Box, Paper, TextField, Typography } from '@mui/material';
 
-export default function Form() {
-  const { register } = useForm();
+type FormValues = {
+  bill: number;
+};
+interface FormProps {
+  billValue: number;
+  setTotalResultPesos: (result: number) => void;
+}
+
+export default function Form({ billValue, setTotalResultPesos }: FormProps) {
+  const { register, handleSubmit } = useForm<FormValues>();
+  const [result, setResult] = useState<number | null>(null);
+  const onSubmit = (data: FormValues) => {
+    const RESULT: number = data.bill * billValue;
+
+    setResult(RESULT);
+    setTotalResultPesos(RESULT);
+  };
 
   return (
-    <form>
-      <label htmlFor='username'>
-        <Typography>Username</Typography>
-        <input {...register('username')} type='text' id='username' />
-      </label>
-    </form>
+    <Box sx={{ display: 'flex', marginTop: '2rem' }}>
+      <form onChange={handleSubmit(onSubmit)} noValidate>
+        <TextField
+          id={billValue.toString()}
+          label={`$${billValue}`}
+          sx={{ width: '12.8rem' }}
+          variant='outlined'
+          {...register('bill', {
+            valueAsNumber: true,
+          })}
+        />
+      </form>
```

Comment:
This is wrong, a single form should be used for all inputs. Not one form per input.